workspaces/liferay-sample-workspace/client-extensions/liferay-sample-data-set-filter/src/index.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+
+// These definitions will ultimately be referenced by
+//
+// import type {FDSFilter} from '@liferay/js-api/data-set';
+
+interface FDSFilterArgs {
+	id: string;
+	onFilterChange: (string) => void;
```

Comment:
`onFilterChange` looked weird to me... Why is DXP passing me an event handler? Shouldn't be me who creates event handlers? :thinking: 

The name is probably confusing... From the code it looks like it is more like a setter for the filter, isn't it?

Reply:
Yeah we can call it `setFilter`, the only hangup with that was the typical React naming conventions. `set*` is usually from useState and `on*` is typically for a callback

Reply:
Yeah, but that's from the POV of the internals, from the POV of the user who's using the API, it's a setter or something equivalent. It's something that is changing the state of the caller. The fact that it is internally used as a callback is a detail implementation which shouldn't leak through the API.

Am I correct? :thinking:

Reply:
Yeah that sounds fine to me, I agree it sounds a bit odd, especially in a non-react context.

I will update